src/compiler/nvcc.rs

```diff
@@ -401,25 +407,45 @@ pub fn generate_compile_commands(
         .unwrap()
         .path;
 
+    // Build nvcc's internal files in `$TMPDIR/$hash_key` so the paths are
+    // stable across compilations. This is important because this path ends
+    // up in the preprocessed output, so using random tmpdir paths leads to
+    // erroneous cache misses.
+    let out_dir = env::temp_dir().join("sccache_nvcc").join({
+        // Combine `hash_key` with the output path in case
```

Comment:
Can we rename `hash_key` to something less ambiguous? It's temp dir subdir stabilizer, `hash_key` is not encoding that, it also needs docs in the `trait` definition.

Reply:
I called it `hash_key` because it's the key produced by the call to `generate_hash_key()` for this `nvcc` compilation. I am fine calling it something else, it just seemed the most descriptive and consistent name.

Reply:
Done